Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1851,6 +1854,39 @@ class BrowserViewController: UIViewController, SearchBarLocationProvider, Themea
 
     // MARK: Autofill
 
+    private func creditCardInitialSetupTelemetry() {
+        // Initial telemetry for credit card autofill status
+        let userDefaults = UserDefaults.standard
+        let key = PrefsKeys.KeyAutofillCreditCardStatus
+        // Default value is true for autofill credit card input
+        let autofillStatus = userDefaults.value(forKey: key) as? Bool ?? true
+        TelemetryWrapper.recordEvent(
+            category: .action,
+            method: .tap,
+            object: .creditCardAutofillEnabled,
+            extras: [
+                TelemetryWrapper.ExtraKey.isCreditCardAutofillEnabled.rawValue: autofillStatus.description
+            ]
+        )
+
+        // Note: Added a 2 second delay so that profile is setup by the team we
+        // request for has account
+        DispatchQueue.main.asyncAfter(deadline: .now() + 2) { [weak self] in
+            guard let self = self else { return }
+            guard self.profile.hasAccount() else { return }
```

Comment:
@lougeniaC64 also, is there a callback that I can use instead of doing the timer for assuming when account is ready?

Reply:
Note: Marking this as draft because of above comments related to fetching rust creditcard status for telemetry

Reply:
Done, resolved!!